File: python/hackerrank/apple_and_orange.py
# https://www.hackerrank.com/challenges/apple-and-orange/problem

from collections import defaultdict

def countApplesAndOranges(s, t, a, b, apples, oranges):
    house_range = defaultdict(bool)
    for i in range(s, t+1):
        house_range[i] = True

    apple_locations =[a + apple for apple in apples]
    orange_locations = [b + orange for orange in oranges]

    apple_count = len([apple for apple in apple_locations if house_range[apple]])
    orange_count = len([orange for orange in orange_locations if house_range[orange]])

    print(apple_count)
    print(orange_count)

File: go/hackerrank/appleAndOrange.go
// https://www.hackerrank.com/challenges/apple-and-orange/problem

func countApplesAndOranges(s int32, t int32, a int32, b int32, apples []int32, oranges []int32) {
    h := map[int32]bool{}

    for i := s; i <= t; i++ {
        h[i] = true
    }

    ca:= 0
    for _, apple := range apples {
        if h[apple + a] {
            ca++
        }
    }
    fmt.Println(ca)

    co := 0
    for _, orange := range oranges {
        if h[orange + b] {
            co++
        }
    }
    fmt.Println(co)
}
